controlplane/kubeadm/controllers/remediation.go

```diff
@@ -190,17 +190,8 @@ func (r *KubeadmControlPlaneReconciler) canSafelyRemoveEtcdMember(ctx context.Co
 		"currentTotalMembers", currentTotalMembers,
 		"currentMembers", etcdMembers)
 
-	// The cluster MUST have at least 3 members, because this is the smallest cluster size that allows any etcd failure tolerance.
-	//
-	// NOTE: This should not happen given that we are checking the number of replicas before calling this method, however
-	// given that this could be destructive, this is an additional safeguard.
-	if currentTotalMembers < 3 {
-		log.Info("etcd cluster with less of 3 members can't be safely remediated")
-		return false, nil
-	}
-
-	targetTotalMembers := currentTotalMembers - 1
-	targetQuorum := targetTotalMembers/2.0 + 1
+	// Projects the target etcd cluster after remediation, considering all the etcd members except the one being remediated.
+	targetTotalMembers := 0
```

Comment:
Isn't this the current total members?

Reply:
No, because the currentTotalMembers might or might not include a member for the machine being remediated, so we are required to loop into all the member and add to the total only what is relevant for the target cluster.